eventuals/type-traits.h

```diff
@@ -235,6 +239,9 @@ struct tuple_types_union<
 template <typename Left, typename Right>
 using tuple_types_union_t = typename tuple_types_union<Left, Right>::type;
 
+template <typename Left, typename Right>
+using tuple_types_unique_t = typename tuple_types_union<Left, Right>::unique_left;
```

Comment:
Let's triple check that anything we add we use and test. Please keep this issue unresolved until we're about to commit this PR, thank you!